make_UPD_igv.py

```diff
@@ -0,0 +1,189 @@
+        if mother != '0':
+            samples[sample]['parents'].append(mother)
+            samples[mother]['children'].append(sample)
+
+    families = {}
+    for sample in samples:
+        if len(samples[sample]['parents']) == 2:
+            families[sample] = samples[sample]['parents']
+
+    return samples, families
+
+
+def parse_vcf(vcf_file):  # returns list with genotypes
+    with open(vcf_file, 'r') as vcf_input_file:
+        vcf_reader = vcf.Reader(vcf_input_file)
+
+        if 'fileformat' not in vcf_reader.metadata: # Check if true VCF file
+            sys.exit("Input file {} is not a correct VCF file. "\
+                "Field \"fileformat\" was not detected in VCF".format(args.inputfile)
+                )
+        if len(vcf_reader.samples) > 1: # Check is VCF is not a multisample VCF
+            sys.exit("Single sample VCF support only. Input file {} is a multisample VCF".format(args.inputfile)) 
+
+        sampleid = vcf_reader.samples[0]
+        snv_list = []
+        for record in vcf_reader:
+            chrom = record.CHROM
+            pos = record.POS
+
+            if 'DP' in record.genotype(sampleid).data._asdict():
+                dp = record.genotype(sampleid)['DP']
+            else: # Skip variant position if DP is missing
+                continue
+
+            if dp >= args.mindepth and chrom is not "Y" :
+                if record.genotype(sampleid).phased: # sort genotype in unphased state.
+                    gt = record.genotype(sampleid)['GT'].split("|")
+                    gt.sort()
+                    gt = "/".join(gt)
+                else:
+                    gt = record.genotype(sampleid)['GT']
+                if len(record.ALT) > 1: # skip multiallelic position
+                   continue
+                allele = [str(record.REF[0]), str(record.ALT[0])]
+                genotype_call = record.genotype(sampleid).is_variant
+                if genotype_call is not None: # True = ref/alt or alt/alt genotype, False = ref/ref genotype, None = missing genotype (i.e. ./.). Skip missing genotypes in statement.
+                    genotype = [] 
+                    for item in gt.split("/"):
+                        genotype.append(allele[int(item)])
+                    allele_genotype = "/".join(genotype)
+                    snv_list.append(["{}_{}".format(chrom, pos), gt])
+    return snv_list
+
+
+def make_upd(families, samples):
+    vcfs = {}
+    for vcf in args.input_files:
+        sampleid = vcf.split(args.suffix)[0].split("/")[-1].strip("_dedup.realigned")
+        if sampleid not in vcfs:
+            vcfs[sampleid] = vcf  
+    for sample in families:
+        family = samples[sample]['family']
+
+        if sample not in vcfs:  # PED file includes all samples of entire run, not only project. Skip if child VCF is not present in current analysis.
+            continue
+    
+        if args.sample_id not in sample:
+            continue
```

Comment:
Is it possible to merge these if statements using `or` ?

Reply:
Yes that makes sense!